compiler/rustc_session/src/config.rs

```diff
@@ -2595,6 +2597,23 @@ fn parse_remap_path_prefix(
     mapping
 }
 
+fn parse_logical_env(
+    handler: &mut EarlyErrorHandler,
+    matches: &getopts::Matches,
+) -> FxHashMap<String, String> {
+    let mut vars = FxHashMap::default();
+
+    for arg in matches.opt_strs("env") {
+        if let Some((name, val)) = arg.split_once('=') {
+            vars.insert(name.to_string(), val.to_string());
```

Comment:
What if the same env is specified multiple times? This code just overrides it without an error. Is that desired?

Reply:
It is desired but it's up to debate. I followed the same logic as:

```terminal
rustc --cap-lints=warn --cap-lints=error foo.rs
```

which doesn't emit a warning or anything.